contracts/games/session-activity/SessionActivityDeployer.sol

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: Apache 2
+// solhint-disable not-rely-on-time
+
+pragma solidity ^0.8.19;
+
+import {AccessControlEnumerable} from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import {SessionActivity} from "./SessionActivity.sol";
+
+error Unauthorized();
+error NameAlreadyRegistered();
+
+/**
+ * @title SessionActivityDeployer - A factory contract that deploys SessionActivity contracts and tracks their addresses and names
+ * @author Immutable
+ * @dev The SessionActivityDeployer contract is not designed to be upgradeable or extended.
+ */
+contract SessionActivityDeployer is AccessControlEnumerable {
+    /// @notice Indicates that an account has registered session activity
+    event SessionActivityDeployed(address indexed account, string indexed name);
+
+    /// @notice Mapping of deployed SessionActivity contract addresses to their names
+    /// @dev To get a list of all deployed contract names, iterate over the deployedContracts array and use this mapping
+    mapping(address deployedContract => string name) public sessionActivityNames;
+
+    /// @notice Mapping of SessionActivity contract names to their addresses
+    /// @dev To get a list of all deployed contract addresses, iterate over the names array and use this mapping
+    mapping(string name => address deployedContract) public sessionActivityContracts;
+
+    /// @notice Array of deployed SessionActivity contracts
+    address[] public deployedContracts;
+
+    /// @notice Array of deployed SessionActivity contract names
+    string[] public names;
+
+    /// @notice Role to allow deploying SessionActivity contracts
+    bytes32 private constant _DEPLOYER_ROLE = keccak256("DEPLOYER");
+
+    /// @notice The address for the pauser role on the SessionActivity contract
+    address private _pauser;
+
+    /// @notice The address for the unpauser role on the SessionActivity contract
+    address private _unpauser;
```

Comment:
Looking for specific feedback and suggestions for effective role management for the Factory + deployed contracts.

Reply:
would the same pauser and unpauser be used all the time? 
What happens is the pauser or unpauser account are compromised?

Reply:
Maybe have setPauserAndUnpauser() function?